notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
+    "                                 (train_data_copy['city'] == city)].groupby('date').mean()['sales']\n",
+    "      trend = average_sales.rolling(\n",
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
+    "  plt.title(family_value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(7,5))\n",
+    "sns.countplot(x=train_data['type'], alpha=0.7, data=train_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 100))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(17, 2, i+1)\n",
+    "  for store_type in train_data_copy['type'].unique():\n",
+    "      #plt.plot(np.arange(len(current_type_and_family_data.index)), current_type_and_family_data['sales'])\n",
+    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
+    "                                 (train_data_copy['type'] == store_type)].groupby('date').mean()['sales']\n",
+    "      trend = average_sales.rolling(\n",
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
+    "  plt.title(family_value)\n",
+    "  plt.legend(train_data_copy['type'].unique())\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>As we can see, the store type has a great impact on the target (type D is leader and it is logical, because the plot above shows us, that type D stores are the most), that is why 'type' feature is very important.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(7,5))\n",
+    "sns.countplot(x=train_data['cluster'], alpha=0.7, data=train_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "sales_and_cluster_data = train_data_copy.drop(['date', 'family', 'onpromotion', 'city', 'type'], axis=1)\n",
+    "sales_and_cluster_data.corr()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figsize=(10, 7)\n",
+    "plt.scatter(x='cluster', y='sales', data=sales_and_cluster_data)"
```

Comment:
It is better to use `boxplot` or `violinplot` here, see [pandas](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.boxplot.html), [matplotlib](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.boxplot.html) and [seaborn](https://seaborn.pydata.org/generated/seaborn.violinplot.html) docs